core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -387,13 +392,68 @@ long getValueReference(int entryIndex) {
         return getEntryFieldLong(entryIndex, OFFSET.VALUE_REFERENCE);
     }
 
+    long getValueReferenceAndVersion(int entryIndex, int[] version) {
+        long valueReference;
+        int v;
+        do {
+            v = getValueVersion(entryIndex);
+            valueReference = getValueReference(entryIndex);
+        } while (v != getValueVersion(entryIndex));
+        version[0] = v;
+        return valueReference;
+    }
+
+    int completeLinking(LookUp lookUp) {
+        int entryVersion = lookUp.version;
+        // no need to complete a thing
+        if (entryVersion > INVALID_VERSION) {
+            return entryVersion;
+        }
+        if (!publish()) {
+            return INVALID_VERSION;
+        }
+        try {
+            Slice valueSlice = buildValueSlice(lookUp.valueReference);
+            int offHeapVersion = operator.getOffHeapVersion(valueSlice);
+            casEntriesArrayInt(lookUp.entryIndex, OFFSET.VALUE_VERSION, entryVersion, offHeapVersion);
+            lookUp.version = offHeapVersion;
+            return offHeapVersion;
+        } finally {
+            unpublish();
+        }
+    }
+
+    NovaValueUtils.NovaResult finalizeDeletion(LookUp lookUp) {
```

Comment:
Yet another Result? Is there an option to collide it all together?

Reply:
Yes, I'll do that!

Reply:
This Result replaces the Result of ValueUtils (It is used inside the compound class Result<V>), so no need to consolidate it.